subgraphs/_reference_/src/sdk/protocols/bridge/tokens.ts

```diff
@@ -0,0 +1,99 @@
+import { Address, BigInt, Bytes, log } from "@graphprotocol/graph-ts";
+import { RewardToken, Token } from "../../../../generated/schema";
+import { CrosschainToken, SupportedToken } from "../../../../generated/schema";
+import { chainIDToNetwork } from "./chain-ids";
+import { Bridge } from "./protocol";
+import { RewardTokenType } from "../../util/constants";
+
+export interface TokenInitializer {
+  getTokenParams(address: Address): TokenParams;
+}
+
+export class TokenParams {
+  name: string;
+  symbol: string;
+  decimals: i32;
+}
+
+export class TokenManager {
+  protocol: Bridge;
+  initializer: TokenInitializer;
+
+  constructor(protocol: Bridge, init: TokenInitializer) {
+    this.protocol = protocol;
+    this.initializer = init;
+  }
+
+  initToken(address: Address): Token {
```

Comment:
I don't have a better naming suggestion atm, but can we break it into `getToken` and `initToken` or just use the `getOrCreateToken` name we have been using? I don't expect to call initToken to return a token (already existing).

Reply:
Sure, yeah. Better to be consistent with what we already have 👍